python/lsst/cell_coadds/_cellCoaddBuilder.py

```diff
@@ -0,0 +1,385 @@
+        to completely overlap the outer bounding box of the cells. Any further
+        exclusion of images based on quality assessment or other criteria
+        should be dome in this method.
+
+        Parameters
+        ----------
+        inputs: `Mapping[ObservationIdentifiers, Tuple[DeferredDatasetHandle,
+                                                       lsst.geom.Box2I]]`
+            A mapping from `lsst.cell_coadds.ObservationIdentifiers`` to a
+            tuple containing a `DeferredDatasetHandle` pointing to the input
+            image (calexp or warps) and a minimal bounding box that can be read
+            without loading the entire image.
+        cellInfo: `lsst.skymap.CellInfo`
+            An `lsst.skymap.CellInfo` object with the following
+            attributes:
+            - wcs: `lsst.afw.geom.SkyWcs`
+            - outer_bbox: `lsst.geom.Box2I`
+
+        Returns
+        -------
+        A pipeBase.Struct object with the following fields:
+        image_planes, psf and inputs.
+        """
+        raise NotImplementedError()
+
+    registry = pexConfig.makeRegistry(doc="Internal registry")
+
+
+singleCellCoaddBuilderRegistry = pexConfig.makeRegistry(doc="Registry of single cell coadd builders")
+
+
+class MultipleCellCoaddBuilderConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=("tract", "patch", "band", "skymap"),
+    defaultTemplates={"inputCoaddName": "deep", "outputCoaddName": "deep", "warpType": "direct"},
+):
+    # Since we defer loading of images, we could take in both calexp and
+    # warps as inputs. Unless, we don't want a dependency on warps existing.
+    # The type of image will be specified in MultipleCellCoaddConfig.
+    calexps = cT.Input(
+        doc="Input exposures to be resampled and optionally PSF-matched onto a SkyMap projection/patch",
+        name="calexp",
+        storageClass="ExposureF",
+        dimensions=("instrument", "visit", "detector"),
+        deferLoad=True,
+        multiple=True,
+    )
+
+    skyMap = cT.Input(
+        doc="Input definition of geometry/box and projection/wcs for coadded exposures",
+        name="skyMap",
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+
+    cellCoadd = cT.Output(
+        doc="Coadded image",
+        name="{outputCoaddName}CellCoaddPickled",
+        storageClass="MultipleCellCoadd",
+        dimensions=("tract", "patch", "skymap", "band", "instrument"),
+    )
+
+
+class MultipleCellCoaddBuilderConfig(
+    pipeBase.PipelineTaskConfig, pipelineConnections=MultipleCellCoaddBuilderConnections
+):
+    """Configuration parameters for the `MultipleCellCoaddBuilderTask`."""
+
+    inputType = pexConfig.ChoiceField(
+        doc="Type of input dataset",
+        dtype=str,
+        allowed={"inputWarps": "Use warps", "calexps": "Use calexps"},
+        default="calexps",
+    )
+
+    psf_dimensions: pexConfig.Field = pexConfig.Field(
```

Comment:
This connection is `snake_case`, while the others are `camelCase`.

Actually, could we just make all of the connections and configs being added here `snake_case`? I think that's what the rest of this package does.

Reply:
I've gotten too comfortable with `camelCase` now. Just to be sure, you meant that only the outward facing names such as config fields, and may be connections, should be in `snake_case` but OK with the variable names used in the code being `camelCase`?